### ISSUE TITLE
Filtering for the device selector doesn't work when multiple devices are allowed

### ISSUE THREAD

**Issue by joshuaspence** [home-assistant/frontend#12246]
### The problem

Filtering devices in the `device` selector (e.g. with `integration`) doesn't seem to be applied in the UI picker when combined with `multiple: true`, but does seem to work with `multiple: false`.

### What version of Home Assistant Core has the issue?

2022.04.0

### What type of installation are you running?

Home Assistant OS

### Link to integration documentation on our website

https://www.home-assistant.io/docs/blueprint/selectors/#device-selector

### Example YAML snippet

```yaml
script:
  test:
    alias: 'Test'
    fields:
      target:
        name: 'Target'
        selector:
          device:
            integration: 'mobile_app'
            multiple: true
    sequence: []
```

### Additional information

![Screenshot from 2022-04-07 11-10-44](https://user-images.githubusercontent.com/234676/162100450-8cce4a71-3da5-481d-b8b3-c899403cc9f5.png)
![Screenshot from 2022-04-07 11-11-48](https://user-images.githubusercontent.com/234676/162100615-50cdc7a1-9deb-45c8-b99a-36daea10fa0b.png)

**Comment by joshuaspence** [comment]
Hmm, I am still seeing this bug in 2022.4.6

**Comment by zsarnett** [comment]
Fix will be in 2022.5
